core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAutoFeatureStep.java

```diff
@@ -404,6 +416,40 @@ public void write(String s, byte[] bytes) {
         file.close();
     }
 
+    private List<String> getResourcePathList(String resourcePattern, String identifier) {
+        try {
+            ArrayList<String> resourcePaths = new ArrayList<>();
+            ClassLoader loader = QuarkusClassLoader.getSystemClassLoader();
+            Enumeration<URL> en = loader.getResources("META-INF");
+            while (en.hasMoreElements()) {
+                URL url = en.nextElement();
+                JarURLConnection urlcon = (JarURLConnection) (url.openConnection());
+                try (JarFile jar = urlcon.getJarFile();) {
```

Comment:
This doesn't seem to work on Multi-Release JARs. Better use SmallRye Common IO's `JarFiles.create` instead